internal/featuretests/v3/cluster_test.go

```diff
@@ -587,35 +588,116 @@ func TestClusterWithHealthChecks(t *testing.T) {
 		WithPorts(v1.ServicePort{Port: 80, TargetPort: intstr.FromString("8080")}),
 	)
 
-	rh.OnAdd(&contour_api_v1.HTTPProxy{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "simple",
-			Namespace: "default",
-		},
-		Spec: contour_api_v1.HTTPProxySpec{
-			VirtualHost: &contour_api_v1.VirtualHost{Fqdn: "www.example.com"},
-			Routes: []contour_api_v1.Route{{
-				Conditions: []contour_api_v1.MatchCondition{{
-					Prefix: "/a",
-				}},
-				HealthCheckPolicy: &contour_api_v1.HTTPHealthCheckPolicy{
-					Path: "/healthz",
+	// proxy1 has a basic health check policy.
+	proxy1 := fixture.NewProxy("default/simple").WithSpec(contour_api_v1.HTTPProxySpec{
+		VirtualHost: &contour_api_v1.VirtualHost{Fqdn: "www.example.com"},
+		Routes: []contour_api_v1.Route{{
+			Conditions: []contour_api_v1.MatchCondition{{
+				Prefix: "/a",
+			}},
+			HealthCheckPolicy: &contour_api_v1.HTTPHealthCheckPolicy{
+				Path: "/healthz",
+			},
+			Services: []contour_api_v1.Service{{
+				Name:   "kuard",
+				Port:   80,
+				Weight: 90,
+			}},
+		}},
+	})
+
+	rh.OnAdd(proxy1)
+
+	c.Request(clusterType).Equals(&envoy_discovery_v3.DiscoveryResponse{
+		Resources: resources(t,
+			clusterWithHealthCheck("default/kuard/80/bc862a33ca", "default/kuard", "default_kuard_80", "/healthz", nil),
+		),
+		TypeUrl: clusterType,
+	})
+
+	// proxy2 has valid expected status ranges.
```

Comment:
I'm experimenting with feature tests only, per discussion this morning, but I can circle back and add more test cases for individual functions if we think that's worthwhile. This does a good job coverage-wise, and is (at least for this feature) more pleasant to write IMO.